Tasks/Xcode/xcodebuild2.js

```diff
@@ -221,3 +229,27 @@ function removeExecOutputNoise(input) {
 	output = output.trim().replace(/[,\n\r\f\v]/gm,'');	
 	return output;
 }
+
+function publishTestResults(publishResults, testResultsFiles) {
+  if(publishResults) {
+    //check for pattern in testResultsFiles
+    if(testResultsFiles.indexOf('*') >= 0 || testResultsFiles.indexOf('?') >= 0) {
+      tl.debug('Pattern found in testResultsFiles parameter');
+      var buildFolder = tl.getVariable('agent.buildDirectory');
+      var allFiles = tl.find(buildFolder);
+      var matchingTestResultsFiles = tl.match(allFiles, testResultsFiles, { matchBase: true });
+    }
+    else {
+      tl.debug('No pattern found in testResultsFiles parameter');
+      var matchingTestResultsFiles = [testResultsFiles];
+    }
+
+    if(!matchingTestResultsFiles) {
```

Comment:
if publish is checked but xctool is not checked give the right warning with the resolution ie enable xctool for publish to work